### DISCUSSION TITLE
Scrollmöglichkeit beim Luftbildvergleich

### DISCUSSION THREAD

**Discussion by nicole-vlbg** [e-netze/webgis-community#85] [Ideas]
Wir vermissen im Luftbildvergleich eine Scroll Möglichkeit zur Auswahl der Luftbilder.
Bei einer grossen Anzahl von Luftbildern und einem kleinen Bildschirm können derzeit nicht alle angezeigt werden

![image](https://github.com/gis-eni/webgis-forum/assets/124671123/d7ec6a40-e3bf-4a2b-9510-1999e5bbd2f4)

**Comment by gis-eni** [comment]
Wie diese Apps aussehen (styling, ...) ist den HTML Templates unter ``portal/_tempates`` definiert. Hier sollten die Vorlagen für die Vergleichsseiten geben ``WebGIS-Service-SideBySide.html`` bzw. ``WebGIS-TileServices-SideBySide.html``.

Es kann auch sein, dass du eine eigene Vorlage verwendest. Dann musst du das in deiner Vorlage machen. Was du gewählt hast, müsstest bei dieser App im ``AppBuilder`` sehen.

Damit man Scrollen kann, kannst du folgendes machen (kommt generell im nächsten Build, kannst aber auch schon gleich nachziehen:
* In beiden oben angeführten Files werden die sog. ``LayerSwitcher" definiert (ca. Zeile 32)
* dort können die Styles für diese Element um ``overflow-y" bzw. ``max-height`` folgendermaßen erweitert werden:

```html
<div id="layerswitch-left" style="position: absolute; left: 5px; top: 5px; overflow-y: auto; max-height: calc(100% - 10px)">
</div>

<div id="layerswitch-right" style="position: absolute; right: 5px; top: 5px; overflow-y: auto; max-height: calc(100% - 10px)">
</div>
``` 

Wenn man jetzt die App das nächste Mal öffnet, wird die Seite mit denen Styles geladen und man sollte in der Liste scrollen können.

**Comment by gis-eni** [comment]
Tipp: Side by Side über benutzerdefiniertes Werkzeug in die Karte einbinden:

``` javascript

webgis.custom.tools.add({
    name: 'TOR Befliegungen',
    command: 'https://localhost:44320/App/PortalApp/dev?app=SideBySide&category=aaa&center={map.centerx},{map.centery}&scale={map.scale}',
    command_target: 'dialog',
    image: '',
    container: "Abfragen"
});

```
